embed-code-go/test/fragmentation_test/fragmentation_test.go

```diff
@@ -0,0 +1,256 @@
+	fragmentChildren, _ := os.ReadDir(config.FragmentsDir)
+	if len(fragmentChildren) != 1 {
+		t.Errorf("Expected 1, got %d", len(fragmentChildren))
+	}
+	if fragmentChildren[0].Name() != "org" {
+		t.Errorf("Expected 'org', got '%s'", fragmentChildren[0].Name())
+	}
+
+	fragmentFiles, _ := os.ReadDir(fmt.Sprintf("%s/org/example", config.FragmentsDir))
+	if len(fragmentFiles) != 4 {
+		t.Errorf("Expected 4, got %d", len(fragmentFiles))
+	}
+
+	defaultFragmentExists := false
+	for _, file := range fragmentFiles {
+		if file.Name() == fileName {
+			defaultFragmentExists = true
+		} else if matched, _ := regexp.MatchString(`Hello-\w+\.java`, file.Name()); !matched {
+			t.Errorf("File name does not match pattern: %s", file.Name())
+		}
+	}
+
+	if !defaultFragmentExists {
+		t.Errorf("Default fragment '%s' does not exist", fileName)
+	}
+}
+
+func TestFailNotOpenFragment(t *testing.T) {
+	testPreparator := newFragmentationTestsPreparator()
+	testPreparator.setup()
+	defer testPreparator.cleanup()
+
+	var configuration = buildTestConfig()
+	path := fmt.Sprintf("%s/org/example/Unopen.java", configuration.CodeRoot)
+	fragmentation := fragmentation.NewFragmentation(path, configuration)
+	err := fragmentation.WriteFragments()
+	if err == nil {
+		t.Errorf("Expected error, got nil")
+	}
+}
+
+func TestFragmentWithoutEnd(t *testing.T) {
+	testPreparator := newFragmentationTestsPreparator()
+	testPreparator.setup()
+	defer testPreparator.cleanup()
+
+	configuration := buildTestConfig()
+	fileName := "Unclosed.java"
+	path := fmt.Sprintf("%s/org/example/%s", configuration.CodeRoot, fileName)
+	fragmentation := fragmentation.NewFragmentation(path, configuration)
+	err := fragmentation.WriteFragments()
+	if err != nil {
+		t.Errorf("Writing fragments went wrong: %d", err)
+	}
+
+	fragmentDir := fmt.Sprintf("%s/org/example", configuration.FragmentsDir)
+	fragmentFiles, _ := os.ReadDir(fragmentDir)
+	if len(fragmentFiles) != 2 {
+		t.Errorf("Expected 2, got %d", len(fragmentFiles))
+	}
+
+	var fragmentFileName string
+	for _, file := range fragmentFiles {
+		if file.Name() != fileName {
+			fragmentFileName = file.Name()
+			break
+		}
+	}
+
+	fragmentContent, _ := os.ReadFile(fmt.Sprintf("%s/%s", fragmentDir, fragmentFileName))
+	fragmentContentStr := string(fragmentContent)
+
+	re, _ := regexp.Compile(`[.\n\s]+}\n}\n`)
+
+	matched := re.FindStringSubmatch(fragmentContentStr)
+
+	if len(matched) == 0 {
+		t.Errorf("Fragment content does not match pattern: %s", fragmentContentStr)
+	}
+}
+
+func TestFragmentizeEmptyFile(t *testing.T) {
+	testPreparator := newFragmentationTestsPreparator()
+	testPreparator.setup()
+	defer testPreparator.cleanup()
+
+	configuration := buildTestConfig()
+	fileName := "Empty.java"
+	path := fmt.Sprintf("%s/org/example/%s", configuration.CodeRoot, fileName)
+	fragmentation := fragmentation.NewFragmentation(path, configuration)
+	fragmentation.WriteFragments()
+
+	fragmentDir := fmt.Sprintf("%s/org/example", configuration.FragmentsDir)
```

Comment:
Can we extract `fragmentDir ` and `path ` to some constant its repeated again and again in the test code. And lets check in general, all this block looks like duplication among all tests. It looks like in each test we have the same fragmentation instantiation code but with a different filename, probably we extract it?